lib/ansible/modules/cloud/amazon/aws_s3_bucket_facts.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/python
+# Copyright (c) 2017 Ansible Project
+# This is a free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: aws_s3_bucket_facts
+short_description: Lists S3 buckets in AWS
+requirements:
+  - boto3 >= 1.4.4
+  - python >= 2.6
+description:
+    - Lists S3 buckets in AWS
+version_added: "2.4"
+author: "Gerben Geijteman (@hyperized)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Note: Only AWS S3 is currently supported
+
+# Lists all s3 buckets
+- aws_s3_bucket_facts:
+'''
+
+RETURN = '''
+Buckets:
+  description: List of buckets
+  returned: always
+  type: list
+  sample:
+    [{
```

Comment:
Tend to favour YAML data structures rather than JSON style

```
  sample:
    - name: my_bucket
      creation_date: "2017-07-06T15:05:12+00:00"
```